core/aws-core/src/main/java/software/amazon/awssdk/awscore/client/builder/AwsDefaultClientBuilder.java

```diff
@@ -176,7 +176,7 @@ private SdkClientConfiguration finalizeAwsConfiguration(SdkClientConfiguration c
                             .lazyOptionIfAbsent(AwsClientOption.DUALSTACK_ENDPOINT_ENABLED, this::resolveDualstackEndpointEnabled)
                             .lazyOptionIfAbsent(AwsClientOption.FIPS_ENDPOINT_ENABLED, this::resolveFipsEndpointEnabled)
                             .lazyOption(AwsClientOption.DEFAULTS_MODE, this::resolveDefaultsMode)
-                            .lazyOption(SdkClientOption.DEFAULT_RETRY_MODE, this::resolveDefaultRetryMode)
+                            .lazyOptionIfAbsent(SdkClientOption.DEFAULT_RETRY_MODE, this::resolveDefaultRetryMode)
```

Comment:
Do we remember the reason why we initially chose to do a `lazyOption` ?

Reply:
Matt would have the background on this, I don't

Reply:
I don't recall.